Segments/Kesmai.mapproj

```diff
@@ -99075,35 +99096,47 @@ return orc;
+            new StockEntry("sword",         () => new ShortSword(),     200, Utility.RandomBetween(1, 4), null),
+            new StockEntry("spear",         () => new Spear(),          150, Utility.RandomBetween(1, 4), null),
+            new StockEntry("mace",          () => new SteelMace(),      250, Utility.RandomBetween(1, 4), null),
+            new StockEntry("rapier",        () => new Rapier(),         200, Utility.RandomBetween(1, 4), null),
+            new StockEntry("axe",           () => new BattleAxe(),      300, Utility.RandomBetween(1, 4), null),
+            new StockEntry("staff",         () => new WoodenStaff(),    150, Utility.RandomBetween(1, 4), null),
+            new StockEntry("greatsword",    () => new Greatsword(),     400, Utility.RandomBetween(1, 4), null),
+            new StockEntry("crossbow",      () => new Crossbow(),       400, Utility.RandomBetween(1, 4), null),
+            new StockEntry("bow",           () => new Shortbow(),       200, Utility.RandomBetween(1, 4), null),
+            new StockEntry("flail",         () => new WoodenFlail(),    250, Utility.RandomBetween(1, 4), null),
+            new StockEntry("shuriken",      () => new Shuriken(),       200, Utility.RandomBetween(1, 4), null),
+            new StockEntry("katana",        () => new Katana(),         300, Utility.RandomBetween(1, 4), null),
+        };
         
-        if (Utility.RandomBool())
-            shopkeeper.Stock<Spear>("spear", 150, Utility.RandomBetween(1, 4));
-            
-        if (Utility.RandomBool())
-            shopkeeper.Stock<Rapier>("rapier", 200, Utility.RandomBetween(1, 4));
-            
-        if (Utility.RandomBool())
-            shopkeeper.Stock<BattleAxe>("axe", 300, Utility.RandomBetween(1, 4));
-            
-        if (Utility.RandomBool())
-            shopkeeper.Stock<WoodenStaff>("staff", 500, Utility.RandomBetween(1, 4));
+        if (Utility.RandomDouble() <= 0.25)  /* 25% chance to spawn */
+            wares.Add(new StockEntry("silver dagger", () => new SilverDagger(), 3500, 1, onRarePurchase));
 
-        if (Utility.RandomBool())
-            shopkeeper.Stock<Shortbow>("bow", 200, Utility.RandomBetween(1, 4));
+        if (Utility.RandomDouble() <= 0.20)  /* 15% chance to spawn */
+            wares.Add(new StockEntry("throwing hammer", () => new ThrowingHammer(), 1500, 1, onRarePurchase));
             
-        if (Utility.RandomDouble() < 0.2)
-            shopkeeper.Stock<Greatsword>("greatsword", 2000);
-            
-        for (var i = Utility.Random(1, 3); i > 0; i--)
+        /* The vendor won't sell everything though, at least not this time around. */
+        foreach (var stock in wares.Random(6))
+            shopkeeper.Stock(stock);
+
+        /* Of the things they stock, they don't exactly have everything, yet. */
+        for (var i = Utility.Random(3, 5); i > 0; i--)
```

Comment:
Since C# has zero-based indices, do you think it'll be clearer for others if we consider 0 the (inclusive) limit wherever we can?

Reply:
I did it this way as I think the statement gets re-evaluated when checking the limit of the if statement.  So Utility.Random would generate a new number each time the loop triggered.  We would have to introduce a new local variable, but yes, we should try to make it clearer for others.